vector/src/Data/Vector/Primitive/Mutable.hs

```diff
@@ -643,21 +584,83 @@ foldrM' :: (PrimMonad m, Prim a) => (a -> b -> m b) -> b -> MVector (PrimState m
 {-# INLINE foldrM' #-}
 foldrM' = G.foldrM'
 
--- | /O(n)/ Monadic right fold (action applied to each element and its index).
+-- | /O(n)/ Monadic right fold using a function applied to each element and its index.
 --
 -- @since 0.12.3.0
 ifoldrM :: (PrimMonad m, Prim a) => (Int -> a -> b -> m b) -> b -> MVector (PrimState m) a -> m b
 {-# INLINE ifoldrM #-}
 ifoldrM = G.ifoldrM
 
--- | /O(n)/ Monadic right fold with strict accumulator (action applied
--- to each element and its index).
+-- | /O(n)/ Monadic right fold with strict accumulator using a function applied
+-- to each element and its index.
 --
 -- @since 0.12.3.0
 ifoldrM' :: (PrimMonad m, Prim a) => (Int -> a -> b -> m b) -> b -> MVector (PrimState m) a -> m b
 {-# INLINE ifoldrM' #-}
 ifoldrM' = G.ifoldrM'
 
+-- Modifying vectors
```

Comment:
I understand that you are trying to organize the locations of functions in the module, but could you please move them back where they were before (same in other modules). It is impossible to look if there is any diff and also all the immediate git blame is lost on those functions. Also I think their previous location was more appropriate, but I can be persuaded otherwise, If you do have a strong point for it. One way or another such changes are better done as a separate PR, this way it is much easier to review it.

Reply:
I moved the sections to match the order from the export list, but I can revert this change.

Reply:
Yes please. Not only this one, but all of changes that move functions around. I was gonna comment on your previous PR, but I felt bad about putting more work on you. However it is really hard to review a PR when a piece of code simply moves, it is almost impossible to see if there are any extra changes to that code or it was just a copy and paste.

It is in fact a general suggestion, one should always try to minimize git diff when possible, it makes it easier on others.